scripts/generate-release-notes.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -exuo pipefail
+
+# Determine the OS to use the correct version of sed.
+# shellcheck disable=SC2209
+SED=sed
+if [[ $(uname) == "Darwin" ]]; then
+  # Check if gnu-sed is installed.
+  if ! command -v gsed &> /dev/null; then
+      echo "gnu-sed is not installed. Please install it using 'brew install gnu-sed'." >&2
+      exit 1
+  fi
+  SED=gsed
+fi
+
+# Get the latest tag from the git repository.
+latest_tag=$(git describe --tags --abbrev=0)
+
+# Replace the first line of the CHANGELOG.md file with the expected metadata.
+# NOTE: This is done to ensure the latest release header is consistent, and will replace the manual placeholder.
+$SED -i "1s/.*/## v$latest_tag / $(date +'%Y-%m-%d')/" CHANGELOG.md
```

Comment:
Also modified the PR template to reflect this change: https://github.com/kubernetes/kube-state-metrics/pull/2325/files#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35.